pkg/agent/git/clone.go

```diff
@@ -34,6 +41,51 @@ func CloneRepository(config *config.Config, project *serverapiclient.Project, au
 		return err
 	}
 
+	if shouldClonePR(project) {
```

Comment:
@harkiratsm this code should not be here. The PR link should be parsed in the Git Provider.

The agent should only care about if it should clone a branch or SHA. If a PR link is sent to the Git Provider, it must get the branch and SHA from its API and attach it to the `Repository` object.

Reply:
@Tpuljak Here, we checkout a specific PR if a PR number is provided in the repository object. We fetch the corresponding refs/pull reference and proceed to checkout the associated hash.

Reply:
@harkiratsm I know, but that shouldn't work that way.

The agent should only clone a branch or checkout the provided SHA. The Git provider is responsible for parsing the PR number into the branch and sha

Reply:
Got it 👍

Reply:
Agent should fetch too. 
Thoughts?

Reply:
@harkiratsm I understand your intention but this is not maintainable.

Your code would work flawlessly for Github, but, for example for Gitlab, it wouldn't because Gitlab does not create those refs. Gitlab works with `merge_requests`. Because of this, it's best that the respective providers attach the info needed to checkout a branch/SHA and the agent only uses that information.